packages/block-library/src/columns/style.scss

```diff
@@ -31,20 +33,37 @@
 
 	// Add space between columns. Themes can customize this if they wish to work differently.
 	// Only apply this beyond the mobile breakpoint, as there's only a single column on mobile.
+
+	// 2 columns at this breakpoint, and items beyond the second will wrap.
 	@include break-small() {
+		&:nth-child(even) {
+			margin-left: $grid-size-large;
+		}
+
 		&:nth-child(odd) {
-			margin-right: $grid-size-large * 2;
+			margin-right: $grid-size-large;
 		}
+	}
+
+	// All columns fit in one row at this breakpoint.
+	@include break-medium() {
```

Comment:
This `even, child, nth, first, not` reset stuff makes me dizzy :dizzy_face:

Any interest in simplifying this?

Something along these lines maybe:
```scss
	@include break-medium() {

		// Fill in the left and right margin holes left by the previous small breakpoint.
		// First column already has a margin-right and no margin-left.
		&:not(:first-child) {
			margin-left: $grid-size-large;
			margin-right: $grid-size-large;
		}

		// Remove right margin on last column.
		&:last-child {
			margin-right: $grid-size-large;
		}
	}
```